src/RileyMaze/model.js

```diff
@@ -0,0 +1,26 @@
+/* exported RMZModel */
+
+const {GObject} = imports.gi;
+const {ClippyWrapper} = imports.clippyWrapper;
+
+const _propFlags = GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT;
+
+var RMZModel = GObject.registerClass({
+    Properties: {
+        robotADirection: GObject.ParamSpec.string('robotADirection', 'Robot A direction', '',
+            _propFlags, 'up'),
+        robotBDirection: GObject.ParamSpec.string('robotBDirection', 'Robot B direction', '',
+            _propFlags, 'up'),
+        instructionCode: GObject.ParamSpec.string('instructionCode', 'Instruction code', '',
+            _propFlags, ''),
+        levelCode: GObject.ParamSpec.string('levelCode', 'level code', '', _propFlags, ''),
+        rileyPosition: GObject.ParamSpec.string('rileyPosition', 'Riley position', '',
```

Comment:
I think `rileyPosition` and `goalPosition` could be passed along to the game as part of the `levelCode` string, since the player defines them in the code view. There isn't a separate UI element where they can be set, so if you keep them inside `levelCode` then you wouldn't have to do any syncing between their values in the player's code block and in this model.